lib/hydra/derivatives/processors/active_encode.rb

```diff
@@ -0,0 +1,53 @@
+require 'active_encode'
+
+module Hydra::Derivatives::Processors
+  class ActiveEncodeError < StandardError
+    def initialize(status, source_path, errors = [])
+      msg = "ActiveEncode status was \"#{status}\" for #{source_path}"
+      msg = "#{msg}: #{errors.join(' ; ')}" unless errors.empty?
+      super(msg)
+    end
+  end
+
+  class ActiveEncode < Processor
+    class_attribute :timeout
+    attr_accessor :encode_class
+
+    def initialize(source_path, directives, opts = {})
+      super
+      @encode_class = opts.delete(:encode_class) || ::ActiveEncode::Base
+    end
+
+    def process
+      encode = encode_class.create(source_path, directives)
+      timeout ? wait_for_encode_with_timeout(encode) : wait_for_encode(encode)
+      encode.output.each do |output|
+        output_file_service.call(output, directives)
+      end
+    end
+
+    def wait_for_encode_with_timeout(encode)
+      Timeout.timeout(timeout) { wait_for_encode(encode) }
+    rescue Timeout::Error
+      cleanup_after_timeout(encode)
+    end
+
+    # Wait until the encoding job is finished.  If the status
+    # is anything other than 'completed', raise an error.
+    def wait_for_encode(encode)
+      sleep Hydra::Derivatives.active_encode_poll_time while encode.reload.running?
+      raise ActiveEncodeError.new(encode.state, source_path, encode.errors) unless encode.completed?
+    end
+
+    # After a timeout error, try to cancel the encoding.
+    def cleanup_after_timeout(encode)
+      encode.cancel!
+    rescue => e
```

Comment:
At this point it isn't really defined what errors may be thrown.  Seems like what might need to happen is ActiveEncode's adapters should catch any errors thrown and raise standard errors defined by ActiveEncode.  I believe the default type for rescue is StandardError and I think that is the best that can be done right now.